tiledb/sm/fragment/fragment_metadata.h

```diff
@@ -803,10 +803,12 @@ class FragmentMetadata {
     const auto fragment_timestamp_start = timestamp_range_.first;
     const auto fragment_timestamp_end = timestamp_range_.second;
 
-    return (array_start_timestamp > fragment_timestamp_start &&
-            array_start_timestamp <= fragment_timestamp_end) ||
-           (array_end_timestamp < fragment_timestamp_end &&
-            array_end_timestamp >= fragment_timestamp_start);
+    auto no_overlap = (array_end_timestamp < fragment_timestamp_start) ||
```

Comment:
I think by the point where we have a fragment metadata included in the opened array, it means there is some kind of overlap. We could skip that condition, but add a comment that we are skipping it and why.

Reply:
You could also use the conditional you have in ArrayDirectory::timestamps_overlap?